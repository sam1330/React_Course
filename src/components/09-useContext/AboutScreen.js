import React from "react";

const AboutScreen = () => {

  return (
    <>
      <h1>About</h1>
      <hr />
    </>
  );
};

export default AboutScreen;