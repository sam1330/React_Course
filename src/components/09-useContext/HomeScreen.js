import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const HomeScreen = () => {

  const { user, setUser } = useContext(UserContext);

  return (
    <>
      <h1>Home</h1>
      <hr />

      <pre>
        { JSON.stringify(user, null, 3) }
      </pre>

      <button
        className="btn btn-outline-danger"
        onClick={() => setUser({})}
      >
        Logout
      </button>
    </>
  );
};

export default HomeScreen;