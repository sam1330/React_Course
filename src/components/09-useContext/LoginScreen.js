import React, { useContext } from "react";

import { UserContext } from "./UserContext";

const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

  return (
    <>
      <h1>Login</h1>
      <hr />

      <button
        className="btn btn-outline-primary"
        onClick={() => setUser({
          id: 1,
          name: "Sammy",
          email: "sammy1301@gmail.com"
        })}
      >
        Login
      </button>
    </>
  );
};

export default LoginScreen;