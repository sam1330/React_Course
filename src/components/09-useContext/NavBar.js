import React from "react";
import { Link, NavLink } from "react-router-dom";

const NavBar = () => {

  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <Link className="navbar-brand" to="/">useContext</Link>
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <NavLink to="/" className="nav-link">Home</NavLink>
          <NavLink to="/about" className="nav-link">About</NavLink>
          <NavLink to="/login" className="nav-link">Login</NavLink>
        </div>
      </div>
    </nav>
  );
};

export default NavBar;