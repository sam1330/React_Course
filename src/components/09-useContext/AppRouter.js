import React from "react";
import {
  BrowserRouter,
  Routes,
  Route
} from "react-router-dom";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import LoginScreen from "./LoginScreen";
import NavBar from "./NavBar";


const AppRouter = () => {

  return (
    <BrowserRouter>
      <div>

        <NavBar />

        <Routes>
          <Route path="/" element={<HomeScreen />} />
          
          <Route path="/about" element={<AboutScreen />} />
          
          <Route path="/login" element={<LoginScreen />} />
          
          <Route path="*" element={ <HomeScreen /> }/>
        </Routes>
      </div>
    </BrowserRouter>
  );
};

export default AppRouter;