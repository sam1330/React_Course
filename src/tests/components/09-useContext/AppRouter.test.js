import React from "react";
import { mount } from "enzyme";
import AppRouter from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("<AppRouter /> tests", () => {

  const user = {
    name: "Sammy",
    email: "sammy13@g.com"
  };

  const wrapper = mount(
    <UserContext.Provider value={ { user } }>
      <AppRouter />
    </UserContext.Provider>
  );

  test('should render correctly', () => {

    expect(wrapper).toMatchSnapshot();

  });
});