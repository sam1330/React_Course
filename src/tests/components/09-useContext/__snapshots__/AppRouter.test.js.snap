// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> tests should render correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  Array [
    <h1>
      Home
    </h1>,
    <hr />,
    <pre>
      {
   "name": "Sammy",
   "email": "sammy13@g.com"
}
    </pre>,
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
    >
      Logout
    </button>,
  ]
</div>
`;
